src/Text/SlamSearch/Parser.purs

```diff
@@ -0,0 +1,17 @@
+module Text.SlamSearch.Parser where
+
+import Data.Either
+import Text.Parsing.Parser
+import Text.SlamSearch.Parser.Terms
+import Text.SlamSearch.Parser.Tokens
+import Text.SlamSearch.Parser.Values
+
+parseSearchQuery :: String -> Either ParseError [SearchTerm]
```

Comment:
Like most search grammars, whitespace represents implicit conjunction. That is, "foo bar" means to search for things which have both "foo" AND "bar". While we can encode this top-level conjunction in an array, it may be better to encode it using an ADT. E.g. `SearchAND foo bar`. The reason is that it paves the way to supporting `OR` in the future: "foo OR bar", which would be parsed into `SearchOR foo bar`.